test/usecases/solve/test_scalar.py

```diff
@@ -0,0 +1,33 @@
+from typing import Optional
+
+import numpy as np
+from neuron import h, gui
+from neuron.units import ms
+
+
+def simulate(method: str, rtol: float, dt: Optional[float] = None):
```

Comment:
I don't mind the type hints. If we want to use them, we need to enforce them. Otherwise inevitably they'll become outdated and highly misleading. I remember this happening to me in other BBP code and it's simply not worth it.

A quick test suggests what `mypy` can be `pip install`ed and easily made to spot issues in our code (if we ignore imports).

Therefore, if you want to keep the type hints, please create an issue with the intention of working on it at some point in the near future.

Reply:
I've removed the type hints for now, will open a separate issue to maybe add those in the future.